### PR TITLE
build: add more docker UI (context/dockerfile handling, iidfile, add-host, network)

### PR DESCRIPTION
**Pull request by tiborvass**
This should add same level of support for stdin/remote contexts as the buildkit builder in moby.
Also preliminary support for `--iidfile` (only for docker driver and single-platform queries).
Added support for `--add-host` and `--network`

### CODE REVIEW

File: build/build.go [hunk 1]
```diff
@@ -208,6 +249,9 @@ func Build(ctx context.Context, drivers []DriverInfo, opt map[string]Options, pw
 			mu.Lock()
 			resp[k] = rr
 			mu.Unlock()
+			if opt.ImageIDFile != "" {
+				return ioutil.WriteFile(opt.ImageIDFile, []byte(rr.ExporterResponse["containerimage.digest"]), 0644)
```
Comment (tonistiigi) [inline build/build.go]: Add a switch between manifest digest and this. If manifest digest exists use that.

File: commands/build.go [hunk 2]
```diff
@@ -133,7 +132,18 @@ func buildTargets(ctx context.Context, dockerCli command.Cli, opts map[string]bu
 	defer cancel()
 	pw := progress.NewPrinter(ctx2, os.Stderr, progressMode)
 
-	_, err = build.Build(ctx, dis, opts, pw)
+	resp, err := build.Build(ctx, dis, opts, pw)
+	if err != nil {
+		return err
+	}
+
+	for k, sr := range resp {
```
[the remark this answers is not in the document]
Comment (tiborvass) [reply commands/build.go]: debug will remove

File: build/build.go [hunk 3]
```diff
@@ -208,6 +260,9 @@ func Build(ctx context.Context, drivers []DriverInfo, opt map[string]Options, pw
...
```
Comment (tonistiigi) [inline build/build.go]: this is also not set if no exporter and !moby

File: build/build.go [hunk 4]
```diff
@@ -219,30 +274,109 @@ func Build(ctx context.Context, drivers []DriverInfo, opt map[string]Options, pw
...
+
+	case isLocalDir(inp.ContextPath):
+		target.LocalDirs["context"] = inp.ContextPath
+		switch inp.DockerfilePath {
+		case "-":
+			dockerfileReader = os.Stdin
+		case "":
+			dockerfileDir = inp.ContextPath
+		default:
+			dockerfileDir = filepath.Dir(inp.DockerfilePath)
```
Comment (tonistiigi) [inline build/build.go]: you shouldn't do this as it exposes a directory potentially outside of context. Also this does not work with symlinks.

File: build/build.go (hunk 3 above)
Comment (tonistiigi) [reply build/build.go]: I guess just check if it was present in the response

**Comment by tonistiigi** [comment]
Please handle the remaining comments in follow-up.

**Pull request merged**